ports/nrf/mphalport.c

```diff
@@ -77,3 +77,87 @@ void mp_hal_stdout_tx_strn_cooked(const char *str, mp_uint_t len) {
 void mp_hal_stdout_tx_str(const char *str) {
     mp_hal_stdout_tx_strn(str, strlen(str));
 }
+
+void mp_hal_delay_us(mp_uint_t us)
+{
+    register uint32_t delay __ASM ("r0") = us;
+    __ASM volatile (
+#ifdef NRF51
+            ".syntax unified\n"
+#endif
+        "1:\n"
+        " SUBS %0, %0, #1\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
+        " NOP\n"
```

Comment:
I think this list of NOPs can be made more compact by using a small for loop, or even jumping to the next instruction. For example `b . + 2` takes up 3 cycles on an M0 processor instead of 1 for a `nop` (thus using 1/3 of the instructions).